packages/visualizations/src/components/Map/Choropleth.svelte

```diff
@@ -63,44 +64,74 @@
-        if (newShapes.type === 'geojson' && !newShapes.geoJson) {
+        if (
+            (newShapes.type === ChoroplethShapeTypes.GeoJson && !newShapes.geoJson) ||
+            (newShapes.type === ChoroplethShapeTypes.VectorTiles && !newShapes.url)
+        ) {
             // We don't have everything we need yet
             return;
         }
 
-        if (newShapes.type === 'geojson') {
-            const computeColors = colorShapes(
-                newShapes.geoJson as FeatureCollection,
-                values,
-                newColorScale,
-                emptyValueColor
-            );
-            const coloredShapes = computeColors.geoJson;
-            dataBounds = computeColors.bounds;
+        let valuesToMatch: ChoroplethDataValue[] = values;
+        let colors;
+        let fillColor: string | (string | string[])[] = emptyValueColor;
+
+        if (valuesToMatch.length !== 0) {
+            if (filter) {
+                valuesToMatch = values.filter((value) => filter?.includes(value.x));
+            }
+            dataBounds = getDataBounds(valuesToMatch);
+            colors = mapKeyToColor(valuesToMatch, dataBounds, newColorScales, emptyValueColor);
+            const matchExpression = ['match', ['get', matchKey]];
+            Object.entries(colors).forEach((e) => matchExpression.push(...e));
+            matchExpression.push(emptyValueColor); // Default fallback color
+            fillColor = matchExpression;
+        }
 
+        if (newShapes.type === ChoroplethShapeTypes.GeoJson && newShapes.geoJson) {
             source = {
                 type: 'geojson',
-                data: coloredShapes,
+                data: newShapes.geoJson,
             };
 
             layer = {
                 type: 'fill',
                 layout: {},
                 paint: {
-                    'fill-color': ['get', 'color'],
+                    'fill-color': fillColor,
                     'fill-opacity': 0.8,
-                    'fill-outline-color': '#fff',
+                    'fill-outline-color': DEFAULT_COLORS.ShapeOutline,
                 },
             };
 
-            bbox = turfBbox(newShapes.geoJson);
-        } else {
-            // eslint-disable-next-line no-console
-            console.error('Unknown shapes type', newShapes.type);
+            bbox = bbox || turfBbox(newShapes.geoJson) || VOID_BOUNDS;
+        } else if (newShapes.type === ChoroplethShapeTypes.VectorTiles) {
+            source = {
+                type: 'vector',
+                tiles: [newShapes.url],
+            };
+
+            layer = {
+                type: 'fill',
+                'source-layer': newShapes.layer,
+                layout: {},
+                paint: {
+                    'fill-color': fillColor,
+                    'fill-outline-color': DEFAULT_COLORS.ShapeOutline,
+                    'fill-opacity': [
+                        'case',
+                        ['boolean', ['feature-state', 'hover'], false],
```

Comment:
Why this rule, and if we want something like this, shouldn't it exist for GeoJSON as well?

Also it doesn't seem to have any effect in the Storybook

Reply:
Oh I think that was some experimentations of mine that wasn't supposed to be committed. My bad.,

Reply:
I'll set 0.8 like for GeoJSON and mutualize the style a bit in the code